In [1]:
import pandas as pd
import numpy as np

import matplotlib.pyplot as plt

saw = pd.read_excel('SleepActivityWeight_merged.xlsx')
ahr = pd.read_excel('AverageHeartRatePerHourWithDate.xlsx')
hci = pd.read_excel('HourlyCaloriesIntensity_merged_WithHour.xlsx')

In [36]:
saw.head(5)

Id ActivityDate  TotalSteps  TotalDistance  VeryActiveDistance  \
0  1503960366   2016-04-12       13162           8.50                1.88   
1  1503960366   2016-04-13       10735           6.97                1.57   
2  1503960366   2016-04-15        9762           6.28                2.14   
3  1503960366   2016-04-16       12669           8.16                2.71   
4  1503960366   2016-04-17        9705           6.48                3.19   

   ModeratelyActiveDistance  LightActiveDistance  VeryActiveMinutes  \
0                      0.55                 6.06                 25   
1                      0.69                 4.71                 21   
2                      1.26                 2.83                 29   
3                      0.41                 5.04                 36   
4                      0.78                 2.51                 38   

   FairlyActiveMinutes  LightlyActiveMinutes  ...  Calories   SleepDay  \
0                   13                   328  ...      1985 2016-04-12   
1                   19                   217  ...      1797 2016-04-13   
2                   34                   209  ...      1745 2016-04-15   
3                   10                   221  ...      1863 2016-04-16   
4                   20                   164  ...      1728 2016-04-17   

  TotalSleepRecords  TotalMinutesAsleep  TotalTimeInBed  WeightKg  \
0                 1                 327             346       NaN   
1                 2                 384             407       NaN   
2                 1                 412             442       NaN   
3                 2                 340             367       NaN   
4                 1                 700             712       NaN   

   WeightPounds  BMI  IsManualReport  LogId  
0           NaN  NaN             NaN    NaN  
1           NaN  NaN             NaN    NaN  
2           NaN  NaN             NaN    NaN  
3           NaN  NaN             NaN    NaN  
4           NaN  NaN             NaN    NaN  

[5 rows x 21 columns]

# Distance Travelled by Users

In [3]:
saw['TotalDistance'].mean()

6.01239025034919

## Physical activity of the users 

In [4]:
saw['TotalDistance'].describe()

count    410.000000
mean       6.012390
std        3.043946
min        0.010000
25%        3.592500
50%        6.270000
75%        8.005000
max       17.540001
Name: TotalDistance, dtype: float64

The average bellabeats user travelled a distance of 6km in one day.

The average individual can run 5 miles (8km) in 40 minutes. 

This means that most bellabeats users are below average physically and are using bellabeats to improve their physical condition. (This could also mean that bellabeats users didn't exercise much during the time that the data was recorded)

We can use the data collected to see if the users were actually exercising by analysing the distance travelled while being very active (Meaning running)

In [5]:
saw['VeryActiveDistance'].mean()

1.446219510257971

Most users only travelled 1.4km while being very active.


The average male running speed (for 5km, so long distance running) is 9.5kmph, and the average female running speed is 8kmph 


Therefore, for an average male, 1.4km is the equivalent of running for approximatively 8.8 minutes. 
For an average female, 1.4km is the equivalent of running for approximatively 10.5 minutes

From the data we can conclude bellabeats users are physically below average (or that they weren't very active during the recorded period). This is later confirmed by looking at the average time they were very active for.

We can compare these results to the very active minutes to see if it corrolates 

In [6]:
saw['VeryActiveMinutes'].mean()

25.046341463414635

Individuals were on average very active for 25 minutes per day.

Studies show that at around 15-30 minutes of running, your metabolism will be kick started to start buring some serious amount of fat. So inviduals are in that bracket and are exercising in a way that is making them loose fat. **Is that their goal?**



Confirms that bellabeats users are physically below average. The average bellabeats user is very active for 25minutes per day but only parcours a distance of 1.4km on average per day. While being active for that same amount of time, an average male would travel on average, 4km. An average female would travel on average 3.4km in that same time frame. 

Meaning that bellabeats most likely primary consumer target are people trying to get into shape, people that are just starting their fitness journey.

Bellabeat's marketing strategy could therefore, target people who are bignners in the fitness world / It could target people who are wanting to change and just need that extra push. 

In [7]:
vad = saw['VeryActiveDistance']
vam = saw['VeryActiveMinutes']
correlation = vad.corr(vam)
correlation

0.8488887530399012

With the above function we make sure that the data is correct and that the correlation btw the two studied columns is high. It is.

## Looking at the average for the 50% in the middle

### The activity

In [8]:
saw['VeryActiveMinutes'].describe()
saw['FairlyActiveMinutes'].describe()
saw['LightlyActiveMinutes'].describe()

count    410.000000
mean     216.541463
std       86.705299
min        2.000000
25%      158.000000
50%      208.000000
75%      263.000000
max      518.000000
Name: LightlyActiveMinutes, dtype: float64

In [9]:
saw['VeryActiveMinutes'][saw['VeryActiveMinutes'].between(0,38)]

0      25
1      21
2      29
3      36
4      38
       ..
405    10
406     0
407     0
408     0
409     0
Name: VeryActiveMinutes, Length: 310, dtype: int64

We can select the values from inside the interquartile range (between Q1 and Q3) to see the what are the user habits of the majority of bellabeats users.

The data shows us that 310 users find themselves to be in the majority of users 

In [10]:
saw['VeryActiveMinutes'][saw['VeryActiveMinutes'].between(0,38)].mean()

8.36774193548387

In [11]:
saw['FairlyActiveMinutes'][saw['FairlyActiveMinutes'].between(0,26.75)].mean()

7.537459283387622

In [12]:
saw['LightlyActiveMinutes'][saw['LightlyActiveMinutes'].between(158,263)].mean() 

210.67149758454107

In [13]:
saw['VeryActiveMinutes'][saw['VeryActiveMinutes'].between(0,38)].mean() + saw['LightlyActiveMinutes'][saw['LightlyActiveMinutes'].between(158,263)].mean() + saw['FairlyActiveMinutes'][saw['FairlyActiveMinutes'].between(0,26.75)].mean() 

226.57669880341257

The average amount of time most bella beats users spent being very active was 8.4 minutes per day.

The average amount of time most bella beats users spent being Lightly active was 210.7 minutes per day.

The average amount of time most bella beats users spent being active was 226.5 minutes per day.

Bellabeat's users do spend a large amount of time being ligtly active. Bringing the total time spent active to an average of 226.5 minutes per day. 

From this we can infer that bellabeat's users don't necessarily partake in high intensity activities but they do still tend to move around quite a lot.

This shows that bellabeat products are not used very much by people who are tracking their high intensity workout. It is more used by the average joe who is simply tracking the calories he has lost in a day, and making sure to maintain his shape.

In [14]:
saw['VeryActiveMinutes'].mean() + saw['LightlyActiveMinutes'].mean() + saw['FairlyActiveMinutes'].mean() 

259.5121951219512

Average amount of time spent being active for all Bellabeat users.

## Counting Calories

In [15]:
saw['Calories'].mean()

2389.2951219512197

In [16]:
saw['Calories'].describe()

count     410.000000
mean     2389.295122
std       758.435040
min       257.000000
25%      1841.000000
50%      2207.000000
75%      2920.000000
max      4900.000000
Name: Calories, dtype: float64

The minimum amount of calories burned in one day in our dataset is 257 calories. However, by doing literally nothing, the average human being burns 1800 calories in 24 hours. Therefore, I will not account of any user with calories burned under 1800 in 24 hours as it would mean they probably didn't have the smart device activated.

In [17]:
saw['Calories'][saw['Calories'].between(1800,4900)].mean()

2650.8662420382166

In [18]:
saw['TotalSteps'].mean()

8514.90975609756

The average person burns around 1800 calories a day doing nothing. The average bellabeat user has burns 2650 calories in 24 hours. An excess of 850 calories on average is burned daily by bellabeats users. 

The middle 50% of users spend on average 226.5 (3.7hours) minutes being active. 
The average bellabeat user takes 8515 steps per day.

A 40min walk burns around 300 calories. And 10000 steps daily burns 300 to 400 calories. 
According to google 3 hours of light activity burns around 810 calories.
(The user activity matches the calories burned)

Bellabeat users burn a lot of calories on average, the trend shows that the device isn't really used to track higher intensity activities but used to keep track of calories lost during an average day for an individual who doesn't partake in high intensity activities. 

# Users Sleep and Body

In [19]:
saw['WeightKg'].mean()

64.16857169015067

In [20]:
saw['WeightKg'].describe()

count     35.000000
mean      64.168572
std       13.307268
min       52.599998
25%       61.200001
50%       61.500000
75%       61.900002
max      133.500000
Name: WeightKg, dtype: float64

In [21]:
saw['BMI'].mean()

24.82857121058873

In [22]:
saw['BMI'].describe()

count    35.000000
mean     24.828571
std       4.063695
min      22.650000
25%      23.889999
50%      24.000000
75%      24.170000
max      47.540001
Name: BMI, dtype: float64

In [23]:
saw['TotalMinutesAsleep'].mean()

419.17317073170733

In [24]:
saw['TotalMinutesAsleep'].describe()

count    410.000000
mean     419.173171
std      118.635918
min       58.000000
25%      361.000000
50%      432.500000
75%      490.000000
max      796.000000
Name: TotalMinutesAsleep, dtype: float64

BellaBeat users spend on average 419 minutes asleep (6h 59min) and have a healthy BMI. 

The average Bellabeat user, therefore, isn't out of shape. The Smart device is more commonly used to maintain a healthy lifestyle rather than trying to 'improve' it, get into shape, or by people who are very unhealthy.

It is used by people who are healthy but not very physically active. The marketing strategy should target those types of people. Which are people who probably work a regular job but try to stay in shape by eating properly and go on walks to stay healthy.

## Correlations

In [25]:
BMI = saw['BMI']
Weight = saw['WeightKg']
Sleep = saw['TotalMinutesAsleep']
Calories = saw['Calories']
Steps = saw['TotalSteps']
VeryActive = saw['VeryActiveMinutes']
InBed = saw['TotalTimeInBed']

In [26]:
BMI.corr(Calories)

0.15381095991691326

In [27]:
BMI.corr(Sleep)

-0.06713730252004485

In [28]:
Steps.corr(InBed)

-0.1662319486736155

In [29]:
Steps.corr(Weight)

-0.3748359774668915

In [30]:
Steps.corr(Sleep)

-0.1903439166676317

In [31]:
Steps.corr(BMI)

-0.40387269834971656

These corrolations show us the impact of the users' habits on his health/body/sleep. 

There is a negative correlation between the amount of Steps taken in a day and the amount of sleep had in a day. Meaning that, the more steps taken in one day, the less sleep the user will have. This could be a negative statistic but it could also be explained by either the fact that taking more steps leads to better sleep and therefore, the need to spend less time sleeping; 
it could also be explained by the users who take more steps in the day are generally more active, less 'lazy' and tend to spend less time in bed and more time being active. This is backed up by the correlation between steps and time in bed, the more steps taken in a day, the less amount of time the user spends in bed.

There is also a negative correlation between Steps and BMI, meaning that more steps take the lower the BMI. A high BMI (above 25) is considered unhealthy so lowering that BMI is positive. 

Finally, there is a faint negative correlation between BMI and Sleep, meaning that the higher the BMI, the less sleep a person gets. It can therefore, be argued that a high BMI leads to trouble sleep, and that the reason why the correlation is so low is because individuals who have a lower BMI tend to be healthier, more active, and get better sleep. Therefore, they spend less time in bed sleeping (not because they sleep bad but because they sleep better and live a more active lifestyle). So a high or a low BMI could lead to less sleep, the two being the case cancel out the correlation and it explains why the correlation is so close to 0.  

These statistics could be used a selling points for the product.

# The Other Tables

## Heart Rate

In [37]:
ahr.head(5)

DateTime          Id     Hours  Hour  AverageValue
0 2016-05-08  8877689391  14:00:00    14     73.119835
1 2016-05-01  8877689391  12:00:00    12     67.782609
2 2016-04-22  8877689391  07:00:00     7     69.995215
3 2016-04-13  8877689391  09:00:00     9     62.082742
4 2016-05-04  8877689391  14:00:00    14     67.268041

In [33]:
ahrgroup = ahr.groupby(['Hour']).mean()
ahrgroup.sort_values(['AverageValue'], ascending = False).head(5)

Id  AverageValue
Hour                            
18    5.563010e+09     81.752045
17    5.557283e+09     81.597924
19    5.529603e+09     80.765644
16    5.554325e+09     80.496373
12    5.570633e+09     79.710245

This table shows the hours at which people have the highest heart rate. 

From 5pm to 7pm are the times where people have the highest heart rate. This could mean that those are the time where people generally exercise. This will be backed up by the next table. 

## Calorie Intensity


In [38]:
hci.head(5)

Id  Calories  TotalIntensity  AverageIntensity       Date  \
0  1503960366        81              20          0.333333 2016-12-04   
1  1503960366        61               8          0.133333 2016-12-04   
2  1503960366        59               7          0.116667 2016-12-04   
3  1503960366        47               0          0.000000 2016-12-04   
4  1503960366        48               0          0.000000 2016-12-04   

       Time  Hour  
0  00:00:00     0  
1  01:00:00     1  
2  02:00:00     2  
3  03:00:00     3  
4  04:00:00     4

In [35]:
hcigroup = hci.groupby(['Hour']).mean()
hcigroup.sort_values(['Calories'], ascending = False).head(5)

Id    Calories  TotalIntensity  AverageIntensity
Hour                                                            
17    4.804199e+09  127.586103       22.220544          0.370342
19    4.804199e+09  121.274924       21.039275          0.350655
18    4.804199e+09  119.492447       20.471299          0.341188
16    4.808676e+09  115.915663       18.240964          0.304016
14    4.786521e+09  115.863768       18.872464          0.314541

This table shows that between 5pm and 7pm are the times where people lose the most calories. Which correlates with the times where people have the highest heart rate.

This highlights a trend in activity and, therefore, the time at which the smart device is most useful? Could be used in marketing strategy by working on features that are especially useful for those times?  